website/public/js/controllers/groupsCtrl.js

```diff
@@ -47,6 +47,11 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       group._editing = false;
     };
 
+    $scope.cancelSave = function(group) {
+      group = group.$get();
```

Comment:
That said, I'm not an expert on client side performance.

Also, thanks for the link. I really need to actually read the docs for the ng-resource service. :smile: